testsuite/regressiontests/unused_names.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# unused_names.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+
+# Use encoding-aware Py3 open also in Py2
+if sys.version_info[0] < 3:
+    from io import open
+
+EXIT_SUCCESS = 0
+EXIT_NAME_H_CPP_MISMATCH = 30
+EXIT_UNUSED_NAME = 31
+EXIT_NO_SOURCE = 126
+
+try:
+    source_dir = os.environ['NEST_SOURCE']
+except KeyError:
+    print("Please make NEST_SOURCE environment variable to point to " +
+          "the source tree you want to check!")
+    sys.exit(EXIT_NO_SOURCE)
+
+# Base names of files that contain const Name definitions
+# (i.e. filenames without filename extension)
+names_files = [
+    "librandom/librandom_names",
+    "nestkernel/nest_names",
+    "topology/topology_names",
+]
```

Comment:
I don't think hard-coding the names is a good idea. Why not 
```
name_files = [n[:-2] for n in glob.glob('*/*_names.h')]
```

Reply:
Isn't that a bit dangerous? I can imagine other files to also end in `_names.h` without actually defining `Name`s in the sense that the checker should check for. The current list of files was stable over quite some years and I assume that to be the case for some more...

Reply:
OK.